tests/Constraints/OfPropertiesTest.php

```diff
@@ -223,4 +226,44 @@ public function getInvalidTests()
             )
         );
     }
+
+    public function testNoPrematureAnyOfException()
+    {
+        $schema = json_decode('{
+            "type": "object",
+            "properties": {
+                "propertyOne": {
+                    "anyOf": [
+                        {"type": "number"},
+                        {"type": "string"}
+                    ]
+                }
+            }
+        }');
+        $data = json_decode('{"propertyOne":"ABC"}');
+
+        $v = new Validator();
+        $v->validate($data, $schema, Constraint::CHECK_MODE_EXCEPTIONS);
+        $this->assertTrue($v->isValid());
+    }
+
+    public function testNoPrematureOneOfException()
```

Comment:
You're not seeing things - thanks for spotting that one! Will sort once I'm back at my desk later today.